src/convertToObject.js

```diff
@@ -3,16 +3,45 @@
 /**
  * Implement convertToObject function:
  *
- * Function takes a string with styles (see an example in [stylesString.js](./stylesString.js))
+ * Function takes a string with styles
+ * (see an example in [stylesString.js](./stylesString.js))
  * and returns an object where CSS properties are keys
- * and values are the values of related CSS properties (see an exampl in [test file](./convertToObject.test.js))
+ * and values are the values of related CSS properties
+ * (see an exampl in [test file](./convertToObject.test.js))
  *
- * @param {string} sourceString
+ * @param {string} resultStr
  *
  * @return {object}
  */
 function convertToObject(sourceString) {
-  // write your code here
+  let resultStr = sourceString;
+
+  while (resultStr.includes('\n' || ' :' || ': ' || ' ;' || '  ')) {
+    resultStr = resultStr.replace('\n', '');
+    resultStr = resultStr.replace(' :', ':');
+    resultStr = resultStr.replace(': ', ':');
+    resultStr = resultStr.replace(' ;', ';');
+    resultStr = resultStr.replace('; ', ';');
+    resultStr = resultStr.replace('  ', ' ');
```

Comment:
Read about regular expressions. They can help you avoid such complex code
https://developer.mozilla.org/en-US/docs/Web/JavaScript/Guide/Regular_Expressions
https://regexr.com/